#### ─────────────────────────────────────────────────────────────
#### 📦 모듈 불러오기 (웹 자동화, 다운로드, 마우스/키보드 제어 등)
#### ─────────────────────────────────────────────────────────────

- from selenium import webdriver  # Selenium: 웹 브라우저 자동화 제어용 라이브러리
- from selenium.webdriver.chrome.service import Service  # Chrome 드라이버 서비스 설정용 (옵션 정의에 필요)
- from selenium.webdriver.common.by import By  # 요소 선택을 위한 방법 정의 (예: By.ID, By.CSS_SELECTOR 등)
- from selenium.webdriver.common.keys import Keys  # 키보드 입력 시 사용 (엔터, 화살표 등)
- from selenium.webdriver.chrome.options import Options  # Chrome 실행 시 부가적인 옵션 설정용
- from webdriver_manager.chrome import ChromeDriverManager  # 크롬 드라이버를 자동으로 설치/관리해주는 매니저
- from urllib.request import urlretrieve  # URL에서 파일을 다운로드하고 로컬에 저장할 수 있는 함수
- import time  # 시간 지연을 위한 기본 모듈 (ex. time.sleep)
- import pyautogui  # 마우스, 키보드 자동 제어 라이브러리 (화면 클릭, 입력 등 자동화에 사용)


service = Service(ChromeDriverManager().install())  
- ChromeDriverManager를 통해 최신 크롬 드라이버를 자동으로 설치하고, Service 객체로 래핑하여 Selenium이 사용할 수 있도록 설정

options = Options()  
- Chrome 실행 시 사용할 옵션들을 설정하기 위한 Options 객체 생성

options.add_argument("--headless")  
- 브라우저를 실제로 띄우지 않고 백그라운드(화면 없이)에서 실행 (GUI 없는 환경에서 유용함)

options.add_argument("window-size=1920x1080")  
- 브라우저 창의 크기를 지정 (headless 모드에서는 크기를 명시하지 않으면 화면 요소가 깨질 수 있음)

options.add_argument("--disable-gpu")  
- GPU 하드웨어 가속 기능을 끔 (headless 모드에서 일부 그래픽 관련 이슈 방지용)

options.add_argument("--user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")  
- 브라우저의 사용자 에이전트(User-Agent)를 설정해, 자동화 탐지를 피하거나 특정 환경을 흉내낼 때 사용

options.add_argument("lang=ko_KR")  
- 브라우저 언어를 한국어로 설정

driver = webdriver.Chrome(service=service, options=options)  
- 위에서 설정한 service와 options를 적용하여 Chrome 드라이버 인스턴스 생성


1. 타이틀 이름 불러오기!

In [8]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
import time

service =Service(ChromeDriverManager().install())
options = Options()
options.add_argument("--headless")
options.add_argument("window-size=1920x1080")
options.add_argument("--disable-gpu")
options.add_argument("--user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
options.add_argument("lang=ko_KR")


driver = webdriver.Chrome(service=service, options=options)
driver.get("https://davelee-fun.github.io/")


# element = driver.find_element(By.CSS_SELECTOR,"div.mainheading h1.sitetitle")
element = driver.find_element(By.XPATH,"//div[@class='mainheading']//h1[@class='sitetitle']")

print(element.text.strip())


time.sleep(2)
driver.quit()

Teddy의 선물 블로그


2. 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- 부모요소의 id속성값이 dev-course-list 인 자식요소 class가 cousrse인 데이터만 출력해주세요.

In [26]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By

service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://crawlingtest.netlify.app/")

# elements = driver.find_elements(By.CSS_SELECTOR, "ul#dev_course_list li.course")
# elements = driver.find_elements(By.XPATH,"//ul[@id='dev_course_list']//li[@class='course'] | //li[@class='course maintitle']")
elements = driver.find_elements(By.XPATH, "//ul[@id='dev_course_list']//li[contains(@class, 'course')]")

for element in elements : 
    print(element.text.strip())

time.sleep(1)
driver.quit()

(초급) - 강사가 실제 사용하는 자동 프로그램 소개 [2]
(초급) - 필요한 프로그램 설치 시연 [5]
(초급) - 데이터를 엑셀 파일로 만들기 [9]
(초급) -     엑셀 파일 이쁘게! 이쁘게! [8]
(초급) -     나대신 주기적으로 파이썬 프로그램 실행하기 [7]
(초급) - 파이썬으로 슬랙(slack) 메신저에 글쓰기 [40]
(초급) - 웹사이트 변경사항 주기적으로 체크해서, 메신저로 알람주기 [12]
(초급) - 네이버 API 사용해서, 블로그에 글쓰기 [42]
(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


3. 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- 해당 사이트 내 커리큘럼 일정표 데이터 출력해주세요.

In [29]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
import time

service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://crawlingtest.netlify.app/")

# elements = driver.find_elements(By.TAG_NAME, "td")
# elements = driver.find_elements(By.CSS_SELECTOR, "table tr td")
elements = driver.find_elements(By.XPATH, "//table//tr//td")
for element in elements :
    print(element.text.strip())

time.sleep(1)
driver.quit()

일정
커리큘럼 타이틀
난이도
5.1 ~ 6.15
나만의 엣지있는 블로그 사이트 만들기 (취미로 익히는 IT)
초급
6.16 ~ 7.31
파이썬과 데이터과학 첫걸음 (IT 기본기 익히기)
중급


4. 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://crawlingtest.netlify.app/
- (중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412] 이것만 출력하기

In [43]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.by import By
import time

service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://crawlingtest.netlify.app/")

# elements = driver.find_elements(By.CSS_SELECTOR,"ul#dev_course_list li.course")
elements = driver.find_elements(By.XPATH,"//ul[@id='dev_course_list']//li[@class='course']")
print(elements[-1].text.strip())

time.sleep(1)
driver.quit()

(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


5. 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 클라스 값이 news인 값만 찾아오시오!

In [40]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
import time

service= Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://divjason.github.io/sellenium-test/")
# element = driver.find_element(By.CSS_SELECTOR,"div.message span.news")
element = driver.find_element(By.XPATH,"//div[@class='message']//span[@class='news']")
print(element.text.strip())

time.sleep(1)
driver.quit()

뉴스: 머신러닝 강좌가 나왔어요~~~


6. 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 클라스 값이 news인 값만 찾아오시오!
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로 클릭하기!

In [47]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from webdriver_manager.chrome import ChromeDriverManager
import time

service= Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://divjason.github.io/sellenium-test/")
driver.execute_script("window.scrollTo(0,document.body.scrollHeight)")

element01 = driver.find_element(By.NAME, "username")
element01.clear()
element01.send_keys("error@error.com")

element02 = driver.find_element(By.NAME, "password")
element02.clear()
element02.send_keys("1234")


time.sleep(2)

element03 = driver.find_element(By.CSS_SELECTOR,"input[type='submit']")
element03.click()

driver.quit()

7. 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 클라스 값이 news인 값만 찾아오시오!
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로 클릭하기!
- 클릭 이후 이동한 페이지에서 message라는 클래스 속성값을 가지고 있는 요소의 텍스트 정보를 출력!

In [51]:
# clear() 없이 해도 가능하다면 이 방법이 제일 간결해요!
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from webdriver_manager.chrome import ChromeDriverManager
import time

service= Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://divjason.github.io/sellenium-test/")
driver.execute_script("window.scrollTo(0,document.body.scrollHeight)")

driver.find_element(By.NAME, "username").send_keys("error@error.com")
driver.find_element(By.NAME, "password").send_keys("1234")

time.sleep(2)

driver.find_element(By.CSS_SELECTOR,"input[type='submit']").click()

elements = driver.find_elements(By.CSS_SELECTOR, "div.message")
for _ in elements :
    print(_.text.strip())

driver.quit()

실제 사이트는 https://megaitlab.com/ 입니다
본 페이지는 로그인된 페이지입니다!
뉴스: 머신러닝 강좌가 나왔어요~~~


8. 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링 해오세요.
- https://divjason.github.io/sellenium-test/
- 클라스 값이 news인 값만 찾아오시오!
- input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼 자동으로 클릭하기!
- 해당 페이지에서 로그아웃 버튼응ㄹ 클릭 후 이동한 페이지에서 다시 message 라는 속성값을  가지고 있는 요소와 텍스트 정보를 출력!

In [57]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from webdriver_manager.chrome import ChromeDriverManager
import time

service= Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)

driver.get("https://divjason.github.io/sellenium-test/")
driver.execute_script("window.scrollTo(0,document.body.scrollHeight)")

driver.find_element(By.NAME, "username").send_keys("error@error.com")
driver.find_element(By.NAME, "password").send_keys("1234")

time.sleep(2)

driver.find_element(By.CSS_SELECTOR,"input[type='submit']").click()

elements = driver.find_elements(By.CSS_SELECTOR, "div.message")
for _ in elements :
    print(_.text.strip())


driver.find_element(By.CSS_SELECTOR,"input[type='submit']").click()   

elements = driver.find_elements(By.CSS_SELECTOR, "div.message")
for _ in elements :
    print(_.text.strip())

driver.quit()

실제 사이트는 https://megaitlab.com/ 입니다
본 페이지는 로그인된 페이지입니다!
뉴스: 머신러닝 강좌가 나왔어요~~~
실제 사이트는 https://megaitlab.com/ 입니다
뉴스: 머신러닝 강좌가 나왔어요~~~
